protocol/validator.go

```diff
@@ -1,28 +1,40 @@
 package protocol
 
 import (
-	// "fmt"
-	// "os"
+	"os"
 
-	// "github.com/santhosh-tekuri/jsonschema"
+	"github.com/santhosh-tekuri/jsonschema/v5"
 )
 
+var schemaPaths = map[string]string{
+    "definitions": "../tbdex/hosted/json-schemas/definitions.json",
+    "resource":    "../tbdex/hosted/json-schemas/resource.schema.json",
+    "offering":    "../tbdex/hosted/json-schemas/offering.schema.json",
+    "balance":     "../tbdex/hosted/json-schemas/balance.schema.json",
+    "message":     "../tbdex/hosted/json-schemas/message.schema.json",
+    "order":       "../tbdex/hosted/json-schemas/order.schema.json",
+    "orderstatus": "../tbdex/hosted/json-schemas/orderstatus.schema.json",
+    "Quote":       "../tbdex/hosted/json-schemas/quote.schema.json",
+    "rfq":         "../tbdex/hosted/json-schemas/rfq.schema.json",
+    "close":       "../tbdex/hosted/json-schemas/close.schema.json",
+}
 
-const (
-    DefinitionsPath = "../tbdex/hosted/json-schemas/definitions.json"
-    ResourcePath = "../tbdex/hosted/json-schemas/resource.schema.json"
-    OfferingPath = "../tbdex/hosted/json-schemas/offering.schema.json"
-    BalancePath = "../tbdex/hosted/json-schemas/balance.schema.json"
-    MessagePath = "../tbdex/hosted/json-schemas/message.schema.json"
-    OrderPath = "../tbdex/hosted/json-schemas/order.schema.json"
-    OrderStatusPath = "../tbdex/hosted/json-schemas/orderstatus.schema.json"
-    QuotePath = "../tbdex/hosted/json-schemas/quote.schema.json"
-    RFQPath = "../tbdex/hosted/json-schemas/rfq.schema.json"
-    ClosePath = "../tbdex/hosted/json-schemas/close.schema.json"
-)
+func Compiler() (*jsonschema.Compiler, error) {
+    compiler := jsonschema.NewCompiler()
+    
+    for _, path := range schemaPaths {
+        file, err := os.Open(path)
```

Comment:
This won't work because `os.Open()` uses a physical filesystem. Instead you'll need to use Go's [embedded file support](https://gobyexample.com/embed-directive), in the `../tbdex` directory add a `hosted.go` file that looks something like this:

```
package tbdex

//go:embed hosted
var HostedFiles embed.FS
```

Then use `fs.ReadDir(tbdex.HostedFiles, "hosted")` rater than iterating over the paths above.

Reply:
LMK if you want to pair on this.